src/main/java/org/logstash/plugins/inputs/http/MessageProcessor.java

```diff
@@ -0,0 +1,51 @@
+package org.logstash.plugins.inputs.http;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import org.logstash.plugins.inputs.http.util.RejectableRunnable;
+
+public class MessageProcessor implements RejectableRunnable {
+    private final ChannelHandlerContext ctx;
+    private final FullHttpRequest req;
+    private final String remoteAddress;
+    private final IMessageHandler messageHandler;
+
+    public MessageProcessor(ChannelHandlerContext ctx, FullHttpRequest req, String remoteAddress,
+                            IMessageHandler messageHandler) {
+        this.ctx = ctx;
+        this.req = req;
+        this.remoteAddress = remoteAddress;
+        this.messageHandler = messageHandler;
+    }
+
+    public void onRejection() {
+        /*
+           TODO Should SO_LINGER be used here to force the termination immediately?
```

Comment:
I don't think so.  HTTP 1.1 should try to reuse the same connection. Also the connection should get closed normally since we are gracefully completing the HTTP request/response life-cycle, and the TIME_WAIT to keep the socket open for business is good thing. 

In Beats input i am pretty sure that SO_LINGER is set to zero since it is TCP only and in times past would forcefully close (RST) the connection.